### PR TITLE
Add MSG_NDB_EVENT

### PR DESCRIPTION
**Pull request by pmiettinen**

### CODE REVIEW

File: c/include/libsbp/ndb.h [hunk 1]
```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2015 Swift Navigation Inc.
```
Comment (dt-exafore) [inline c/include/libsbp/ndb.h]: 2016-2017

Comment (dt-exafore) [inline c/include/libsbp/ndb.h]: and similar in other files

File: spec/yaml/swiftnav/sbp/ndb.yaml [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      This message is sent out when an object is stored into NDB. If needed
+      message could also be sent out when fetching an object from NDB.
+    fields:
+      - recv_time:
+          type: u64
+          units: ms
+          desc: HW time in milliseconds.
+      - event:
+          type: u8
+          desc: |
```
Comment (ljbade) [inline spec/yaml/swiftnav/sbp/ndb.yaml]: can you use the `fields` tag to describe this and all other flag/enum type fields?

File: c/include/libsbp/ndb.h [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+ * message could also be sent out when fetching an object from NDB.
+ */
+#define SBP_MSG_NDB_EVENT 0x0400
+typedef struct __attribute__((packed)) {
+  u64 recv_time;      /**< HW time in milliseconds. [ms] */
+  u8 event;          /**< Store = 1,
+Fetch = 2 
+ */
+  u8 object_type;    /**< Ephemeris = 1,
+Almanac = 2,
```
Comment (dt-exafore) [inline c/include/libsbp/ndb.h]: We also store Almanac week number separately. Should we add it here and in other files?

File: spec/yaml/swiftnav/sbp/ndb.yaml (hunk 2 above)
Comment (ljbade) [inline spec/yaml/swiftnav/sbp/ndb.yaml]: also can you explicitly specify what `0` means in all these fields, even if you intend 0 to be marked as "Invalid" or "Reserved"

File: spec/yaml/swiftnav/sbp/ndb.yaml [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+             NDB_DS_SBP
+             Valid only if event is Store.
+      - sid:
+           type: GnssSignal16
+           desc: |
+             GNSS signal identifier,
+             If object_type is Ephemeris OR Almanac, sid indicates for which
+             signal the object belongs to. If object_type is Iono OR L2C
+             capabilities AND data_source is NDB_DS_RECEIVER sid indicates
+             from which SV data was decoded.
```
Comment (ljbade) [inline spec/yaml/swiftnav/sbp/ndb.yaml]: Can you mention the field is reserved for other object types?

File: spec/yaml/swiftnav/sbp/ndb.yaml [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      - sender:
+           type: u16
+           desc: |
+             A unique identifier of the sending hardware. For v1.0,
+             set to the 2 least significant bytes of the device serial
+             number, valid only if data_source is NDB_DS_SBP
```
Comment (ljbade) [inline spec/yaml/swiftnav/sbp/ndb.yaml]: Can you mention this field is reserved for other data sources

File: c/include/libsbp/ndb.h (hunk 1 above)
Comment (ljbade) [inline c/include/libsbp/ndb.h]: This is a generated file, so might need to be updated in the template. I would create a separate issue/PR to fix tho.

Comment (dt-exafore) [inline c/include/libsbp/ndb.h]: @ljbade yeah, I know. Just was too lazy scroll down :)

File: spec/yaml/swiftnav/sbp/ndb.yaml [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+                    - 1: STORE
+                    - 2: FETCH
+      - object_type:
+          type: u8
+          desc: Event object type.
+          fields:
+            - 3-7:
+                desc: Reserved.
+            - 0-2:
+                desc: Event type.
```
Comment (ljbade) [inline spec/yaml/swiftnav/sbp/ndb.yaml]: event object type

**Pull request merged**